comp/otelcol/otlp/config.go

```diff
@@ -83,7 +131,9 @@ func FromAgentConfig(cfg config.Reader) (PipelineConfig, error) {
 	if !metricsEnabled && !tracesEnabled && !logsEnabled {
 		errs = append(errs, fmt.Errorf("at least one OTLP signal needs to be enabled"))
 	}
-	metricsConfig := readConfigSection(cfg, coreconfig.OTLPMetrics)
+	v := serializerexporter.MetricsConfig{}
+	mp := typeTagMap(v, "", map[string]string{})
+	metricsConfig := readConfigSection(cfg, coreconfig.OTLPMetrics, mp)
```

Comment:
I would change this interface to be this instead:

```suggestion
	metricsConfig := readConfigSection(cfg, coreconfig.OTLPMetrics, &serializerexporter.MetricsConfig{})
```

Leaving the responsibility of building the type map as an implementation detail of `readConfigSection`.